sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/UploadFile.cs

```diff
@@ -0,0 +1,116 @@
+        /// <summary>
+        /// Access manager to the Azure storage Share.
+        /// This is shared across all instances of the test run.
+        /// </summary>
+        private static ShareClientManager s_shareClientManager;
+
+        /// <summary>
+        /// The path of the local file uploaded to Azure storage.
+        /// </summary>
+        private string _uploadFilePath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UploadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public UploadFile(SizeOptions options) : base(options)
+        {
+        }
+
+        /// <summary>
+        /// Instantiates and creates a <see cref="ShareClient"/> and a <see cref="ShareDirectoryClient"/> for the test.
+        /// </summary>
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            s_shareClientManager = new ShareClientManager();
+            await s_shareClientManager.CreateShareClientAsync();
+
+            _uploadFilePath = await ShareClientManager.CreateRandomFileAsync(Options.Size);
+        }
+
+        /// <summary>
+        /// Marks the <see cref="ShareClient"/> used by the test for deletion.
+        /// </summary>
+        public override async Task GlobalCleanupAsync()
+        {
+            await base.GlobalCleanupAsync();
+
+            await s_shareClientManager.DisposeAsync();
+        }
+
+        /// <summary>
+        /// Creates a file of the appropriate size filled with random characters.
+        /// The test will use a <see cref="ShareFileClient"/> to upload this file.
+        /// </summary>
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            _uploadFilePath = await ShareClientManager.CreateRandomFileAsync(Options.Size);
+        }
+
+        /// <summary>
+        /// Deletes the local file created by the test.
+        /// </summary>
+        public override async Task CleanupAsync()
+        {
+            await base.CleanupAsync();
+
+            try
+            {
+                File.Delete(_uploadFilePath);
+            }
+            catch
+            {
+            }
+        }
+
+        /// <summary>
+        /// Uploads a file to Azure Shares files storage by calling <see cref="ShareFileClient.Upload(Stream, IProgress{long}, CancellationToken)"/>.
+        /// </summary>
+        /// <param name="cancellationToken">The token used to signal cancellation request.</param>
+        public override void Run(CancellationToken cancellationToken)
+        {
+            ShareFileClient fileClient = s_shareClientManager.DirectoryClient.GetFileClient(Path.GetRandomFileName());
+
+            using (FileStream stream = File.OpenRead(_uploadFilePath))
+            {
+                fileClient.Create(stream.Length, cancellationToken: cancellationToken);
```

Comment:
I think we should be able to call `Create()` once in `Setup()`, then just call `Upload()` in `Run()` to overwrite the file content every time.  We want to measure the perf of just `Upload()`, not `Create() + Upload()`.